### PR TITLE
Adds new flavor to Protein Bars

### PR DESCRIPTION
**Pull request by chesse20**
## About The Pull Request
Adds 8 flavors of protein bars alongside the classic stale protein bar and makes protein bars a grayscaled sprite. 
dreamseeker_XmwY3ehLU9

## Why It's Good For The Game
Marines players love the protein bars but protein bars don't have the variety of sprites and contents of MRE's. This changes this by adding 8 flavors of protein bars each with their own flavor text

## Changelog
:cl:
add: 8 new flavors of protein bars
refactor: made protein bars grayscale
/:cl:

### CODE REVIEW

**Comment by ilikesaying** [comment]
**YESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYES**

**Comment by RipGrayson** [comment]
Probably not in scope for this PR, but none of the custom flavors will work because taste has been broken forever. Eating a bar will get you the "You can taste something indescribable." message, which is the default when food doesn't have a proper flavor.

**Comment by chesse20** [comment]
> Probably not in scope for this PR, but none of the custom flavors will work because taste has been broken forever. Eating a bar will get you the "You can taste something indescribable." message, which is the default when food doesn't have a proper flavor.

yea i know i tested it but atomic prs for 1 and two idk how to fix taste rn

**Comment by ilikesaying** [comment]
> > Probably not in scope for this PR, but none of the custom flavors will work because taste has been broken forever. Eating a bar will get you the "You can taste something indescribable." message, which is the default when food doesn't have a proper flavor.
> 
> yea i know i tested it but atomic prs for 1 and two idk how to fix taste rn

Taste is based on the reagent ingested, MRE food contains the reagent sodium chloride which causes them to taste like salt
pills that contain dylovene will taste like "a roll of gauze" because that's what's stated in the code.

File: code/game/objects/items/marine_gear.dm [hunk 1]
```diff
@@ -53,6 +54,73 @@
...
+	greyscale_config = /datum/greyscale_config/protein
+
+/obj/item/reagent_containers/food/snacks/protein_pack_base/one
+	name = "Stale TGMC protein bar"
+	desc = "The most fake looking protein bar you have ever laid eyes on, covered in the a subtitution chocolate. The powder used to make these is a subsitute of a substitute of whey substitute."
+	greyscale_colors = "#f37d43" //default colors
+	tastes = list("nutraloafed food" = 25)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/two
+	name = "Minty TGMC protein bar"
+	desc = "Stale old protien bar, with an almost minty freshness to it, but not fresh enough"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:66]:
```suggestion
	desc = "A stale old protein bar, with an almost minty freshness to it, but not fresh enough."
```

File: code/game/objects/items/marine_gear.dm [hunk 2]
```diff
@@ -53,6 +54,73 @@
 	list_reagents = list(/datum/reagent/consumable/nutriment = 8)
 	bitesize = 4
 	tastes = list("nutraloafed food" = 7, "cocoa" = 1)
+	greyscale_config = /datum/greyscale_config/protein
+
+/obj/item/reagent_containers/food/snacks/protein_pack_base/one
+	name = "Stale TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:60]:
```suggestion
	name = "stale TGMC protein bar"
```

File: code/game/objects/items/marine_gear.dm [hunk 3]
```diff
@@ -53,6 +54,73 @@
...
+	desc = "The most fake looking protein bar you have ever laid eyes on, covered in the a subtitution chocolate. The powder used to make these is a subsitute of a substitute of whey substitute."
+	greyscale_colors = "#f37d43" //default colors
+	tastes = list("nutraloafed food" = 25)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/two
+	name = "Minty TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:65]:
```suggestion
	name = "mint TGMC protein bar"
```

File: code/game/objects/items/marine_gear.dm [hunk 4]
```diff
@@ -53,6 +54,73 @@
...
+	desc = "Stale old protien bar, with an almost minty freshness to it, but not fresh enough"
+	greyscale_colors = "#61b36e"
+	tastes = list("minty" = 25)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/three
+	name = "Grape TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:70]:
```suggestion
	name = "grape TGMC protein bar"
```

File: code/game/objects/items/marine_gear.dm [hunk 5]
```diff
@@ -53,6 +54,73 @@
...
+	desc = "Not the good type of grape flavor, tastes like medicine. Fills you up just as well as any protein bar."
+	greyscale_colors = "#9900ff"
+	tastes = list("artifical grape" = 25)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/four
+	name = "Mystery TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:75]:
```suggestion
	name = "mystery TGMC protein bar"
```

File: code/game/objects/items/marine_gear.dm [hunk 6]
```diff
@@ -53,6 +54,73 @@
...
+	desc = "Some say they have tasted one of these and tasted their favorite childhood meal. Most say it tastes like crayons"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:76]:
```suggestion
	desc = "Some say they have tasted one of these and tasted their favorite childhood meal, especially for squad marines. Most say this tastes like crayons, though it fills like any other protein bar you've seen."
```

File: code/game/objects/items/marine_gear.dm [hunk 7]
```diff
@@ -53,6 +54,73 @@
...
+	greyscale_colors = "#ffffff"
+	tastes = list("crayons" = 25)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/five
+	name = "Dark Chocolate TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:80]:
```suggestion
	name = "dark chocolate TGMC protein bar"
```

File: code/game/objects/items/marine_gear.dm [hunk 8]
```diff
@@ -53,6 +54,73 @@
...
+	desc = "The dark chocolate flavor helps it out a bit, but its still a cheap protein bar."
+	greyscale_colors = "#5a3b1d"
+/obj/item/reagent_containers/food/snacks/protein_pack_base/six
+	name = "Milk Chocolate TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:84]:
```suggestion
	name = "milk chocolate TGMC protein bar"
```

File: code/game/objects/items/marine_gear.dm [hunk 9]
```diff
@@ -53,6 +54,73 @@
...
+	desc = "A nice milky addition to a otherwise bland protein taste."
+	greyscale_colors = "#efc296"
+	tastes = list("off flavor milk chocolate" = 25)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/seven
+	name = "Raspberry Lime TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:89]:
```suggestion
	name = "raspberry lime TGMC protein bar"
```

File: code/game/objects/items/marine_gear.dm [hunk 10]
```diff
@@ -53,6 +54,73 @@
...
+	desc = "A flavored protein bar, some might say a bit too strongly flavored."
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:90]:
```suggestion
	desc = "A flavored protein bar, some might say a bit too strongly flavored for their tastes."
```

File: code/game/objects/items/marine_gear.dm [hunk 11]
```diff
@@ -53,6 +54,73 @@
...
+	greyscale_colors = "#ff0066"
+	tastes = list("raspberry" = 25,"lime" = 25)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/eight
+	name = "Chicken TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:94]:
```suggestion
	name = "chicken TGMC protein bar"
```

File: code/game/objects/items/marine_gear.dm [hunk 12]
```diff
@@ -53,6 +54,73 @@
...
+	desc = "Protein bar covered with chicken powder one might find in ramen. Get some extra sodium with your protein."
+	greyscale_colors = "#cccc00"
+	tastes = list("chicken and salt" = 25)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/nine
+	name = "Blueberry TGMC protein bar"
```
Comment (XSlayer300) [inline code/game/objects/items/marine_gear.dm:99]:
```suggestion
	name = "blueberry TGMC protein bar"
```

Comment (chesse20) [reply code/game/objects/items/marine_gear.dm]: good fixes

**Comment by chesse20** [comment]
> > > Probably not in scope for this PR, but none of the custom flavors will work because taste has been broken forever. Eating a bar will get you the "You can taste something indescribable." message, which is the default when food doesn't have a proper flavor.
> > 
> > 
> > yea i know i tested it but atomic prs for 1 and two idk how to fix taste rn
> 
> Taste is based on the reagent ingested, MRE food contains the reagent sodium chloride which causes them to taste like salt pills that contain dylovene will taste like "a roll of gauze" because that's what's stated in the code.

Alright i read this comment and slept on it, if i add flavor reagents and make the protein bars contain 2 units of flavor and 8 units of nutriment this should work

**Comment by Pill-U** [comment]
tgmc stale bars memey 2
I like it

**Comment by chesse20** [comment]
image

**Comment by Viniwizi** [comment]
Finally, no eating shit flavor anymore.

**Review by Kuro020 — APPROVED** [review]

File: code/game/objects/items/marine_gear.dm [hunk 13]
```diff
@@ -53,6 +54,74 @@
...
 	bitesize = 4
 	tastes = list("nutraloafed food" = 7, "cocoa" = 1)
+	greyscale_config = /datum/greyscale_config/protein
+
+/obj/item/reagent_containers/food/snacks/protein_pack_base/one
+	name = "stale TGMC protein bar"
+	desc = "The most fake looking protein bar you have ever laid eyes on, covered in the a subtitution chocolate. The powder used to make these is a subsitute of a substitute of whey substitute."
+	greyscale_colors = "#f37d43" //default colors
+	list_reagents = list(/datum/reagent/consumable/protein = 8)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/two
```
Comment (hyper2snyper) [inline code/game/objects/items/marine_gear.dm:64]:
```suggestion

/obj/item/reagent_containers/food/snacks/protein_pack_base/two
```
put an empty line between new object definitions

File: code/game/objects/items/marine_gear.dm [hunk 14]
```diff
@@ -53,6 +54,74 @@
...
+	greyscale_colors = "#cccc00"
+	list_reagents = list(/datum/reagent/consumable/protein/chicken = 8)
+/obj/item/reagent_containers/food/snacks/protein_pack_base/nine
+	name = "blueberry TGMC protein bar"
+	desc = "A nice blueberry crunch into your otherwise stale and boring protein bar."
+	greyscale_colors = "#4e39c5"
+	list_reagents = list(/datum/reagent/consumable/protein/blueberry = 8)
+
+/obj/item/reagent_containers/food/snacks/protein_pack
+
```
Comment (hyper2snyper) [inline code/game/objects/items/marine_gear.dm:107]:
```suggestion
```

File: code/game/objects/items/marine_gear.dm [hunk 15]
```diff
@@ -53,6 +54,74 @@
...
+	name = "TGMC protein bar"
+	desc = "A protien bar produced for the TGMC, comes in many flavors"// this text is only seen at the vending machine, the actual item should never exist.
+/obj/item/reagent_containers/food/snacks/protein_pack/New()
```
Comment (hyper2snyper) [inline code/game/objects/items/marine_gear.dm:110]:
```suggestion

/obj/item/reagent_containers/food/snacks/protein_pack/New()
```
Also use Initialize() here

File: code/modules/reagents/reagents/food.dm [hunk 16]
```diff
@@ -76,6 +76,51 @@
...
+//various flavors of nutriment for protein bars
+/datum/reagent/consumable/protein
+	name = "Protein Mix"
+	description = "All the vitamins, minerals, and carbohydrates the body with a unique flavor"
+	nutriment_factor = 15
+	color = "#664330" // rgb: 102, 67, 48
+	taste_description = "nutraloafed food"
+	var/brute_heal = 1
+	var/burn_heal = 0
+	var/blood_gain = 0.4
```
Comment (hyper2snyper) [inline code/modules/reagents/reagents/food.dm:88]: autodoc

File: code/modules/reagents/reagents/food.dm [hunk 17]
```diff
@@ -76,6 +76,51 @@
...
+		var/mob/living/carbon/C = L
+		if(C.blood_volume < BLOOD_VOLUME_NORMAL)
+			C.blood_volume += blood_gain
+
+	return ..()
+
+/datum/reagent/consumable/protein/mint
+	name = "Mint Protein Mix"
+	taste_description = "minty protein"
+/datum/reagent/consumable/protein/grape
```
Comment (hyper2snyper) [inline code/modules/reagents/reagents/food.dm:103]: new lines between declarations

File: code/modules/reagents/reagents/food.dm [hunk 18]
```diff
@@ -76,6 +76,51 @@
...
+	name = "Protein Mix"
+	description = "All the vitamins, minerals, and carbohydrates the body with a unique flavor"
+	nutriment_factor = 15
+	color = "#664330" // rgb: 102, 67, 48
+	taste_description = "nutraloafed food"
+	var/brute_heal = 1
+	var/burn_heal = 0
+	var/blood_gain = 0.4
+
+/datum/reagent/consumable/protein/on_mob_life(mob/living/L, metabolism)
```
Comment (hyper2snyper) [inline code/modules/reagents/reagents/food.dm:90]: This looks really pretty useful. Is this copied from another food reagent?

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
Formatting mainly

File: code/modules/reagents/reagents/food.dm (hunk 18 above)
Comment (chesse20) [reply code/modules/reagents/reagents/food.dm:90]: yeah its just nutriment, flavor is broken and i can't tell why from the code so this is a band-aid fix

Comment (hyper2snyper) [reply code/modules/reagents/reagents/food.dm:90]: Fair enough

File: code/game/objects/items/marine_gear.dm [hunk 19]
```diff
@@ -53,7 +54,80 @@
...
+/obj/item/reagent_containers/food/snacks/protein_pack_base/nine
+	name = "blueberry TGMC protein bar"
+	desc = "A nice blueberry crunch into your otherwise stale and boring protein bar."
+	greyscale_colors = "#4e39c5"
+	list_reagents = list(/datum/reagent/consumable/protein/blueberry = 8)
+
+/obj/item/reagent_containers/food/snacks/protein_pack
+	name = "TGMC protein bar"
+	desc = "A protien bar produced for the TGMC, comes in many flavors"// this text is only seen at the vending machine, the actual item should never exist.
+/obj/item/reagent_containers/food/snacks/protein_pack/Initialize()
```
Comment (hyper2snyper) [inline code/game/objects/items/marine_gear.dm:117]:
```suggestion

/obj/item/reagent_containers/food/snacks/protein_pack/Initialize()
```

File: code/modules/reagents/reagents/food.dm (hunk 16 above)
Comment (hyper2snyper) [reply code/modules/reagents/reagents/food.dm:88]: autodoc still

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

**Comment by chesse20** [comment]
autodoc should show up now i think

**Pull request merged**